src/prefect/server/database/orm_models.py

```diff
@@ -1209,18 +1246,28 @@ class WorkQueue(Base):
         foreign_keys=[work_pool_id],
     )
 
+    __table_args__ = (
+        sa.UniqueConstraint("work_pool_id", "name"),
+        sa.Index("ix_work_queue__work_pool_id_priority", "work_pool_id", "priority"),
+        sa.Index("trgm_ix_work_queue_name", "name", postgresql_using="gin").ddl_if(
+            dialect="postgresql"
+        ),
+    )
+
 
 class WorkPool(Base):
     """SQLAlchemy model of an worker"""
 
     name = sa.Column(sa.String, nullable=False)
     description = sa.Column(sa.String)
-    type: Mapped[str] = mapped_column(sa.String)
+    type: Mapped[str] = mapped_column(sa.String, index=True)
     base_job_template = sa.Column(JSON, nullable=False, server_default="{}", default={})
     is_paused: Mapped[bool] = mapped_column(
         sa.Boolean, nullable=False, server_default="0", default=False
     )
-    default_queue_id: Mapped[UUID] = mapped_column(UUID, nullable=True)
+    default_queue_id: Mapped[UUID] = mapped_column(
+        UUID, sa.ForeignKey("work_queue.id", ondelete="RESTRICT"), nullable=True
```

Comment:
When I generate a new migration from your branch I see this warning:
```
SAWarning: Cannot correctly sort tables; there are unresolvable cycles between tables
"work_pool, work_queue", which is usually caused by mutually dependent foreign ke
constraints.  Foreign key constraints involving these tables will not be considered;
this warning may raise an error in a future release.`
```
From the [SQLAlchemy docs](https://docs.sqlalchemy.org/en/20/core/constraints.html#creating-dropping-foreign-key-constraints-via-alter), it looks like we can resolve this using the `use_alter` kwarg to resolve the dependency cycle:
```suggestion
        UUID, sa.ForeignKey("work_queue.id", use_alter=True, ondelete="RESTRICT"), nullable=True
```